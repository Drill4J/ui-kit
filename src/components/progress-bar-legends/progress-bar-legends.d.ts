interface Props {
  className?: string;
}
export declare const ProgressBarLegends: {
  ({ className }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
