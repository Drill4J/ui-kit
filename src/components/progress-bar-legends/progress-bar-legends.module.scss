@import '../../style-guide/index';

.progress-bar-legends {
  &__legend {
    position: relative;
    width: 1px;
    height: 8px;
    background-color: map-get($monochrome, dark-tint);
  }

  &__percentage {
    position: absolute;
    left: 50%;
    top: 8px;
    transform: translate(-50%);
    font-size: 12px;
    color: map-get($monochrome, dark-tint);

    &--type {
      &--start {
        left: 4px;
      }
      &--end {
        left: -8px;
      }
    }
  }
}
