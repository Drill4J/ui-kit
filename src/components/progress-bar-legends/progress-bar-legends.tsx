import styled, { css } from 'styled-components';

import { Panel } from '../../layouts';
import { COLORS } from '../../theme';

interface Props {
  className?: string;
}

export const ProgressBarLegends = ({ className }: Props) => (
  <div className={className}>
    <Panel align="space-between">
      <Legend><Percentage type="start">0</Percentage></Legend>
      <Legend><Percentage>25</Percentage></Legend>
      <Legend><Percentage>50</Percentage></Legend>
      <Legend><Percentage>75</Percentage></Legend>
      <Legend><Percentage type="end">100</Percentage></Legend>
    </Panel>
  </div>
);

const Legend = styled.div`
  position: relative;
  width: 1px;
  height: 8px;
  background-color: ${COLORS.MONOCHROME.DARK_TINT};
`;

const Percentage = styled.div<{type?: 'start' | 'end'}>`
  position: absolute;
  left: 50%;
  top: 8px;
  transform: translate(-50%);
  font-size: 12px;
  color: ${COLORS.MONOCHROME.DARK_TINT};
  
  ${({ type }) => [
    type === 'start' && css`left: 4px`,
    type === 'end' && css`left: -8px`,
  ]}
`;
