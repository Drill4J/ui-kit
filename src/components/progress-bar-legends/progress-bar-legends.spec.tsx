import React from 'react';
import renderer from 'react-test-renderer';
import { ProgressBarLegends } from './progress-bar-legends';

describe('ProgressBarLegends', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<ProgressBarLegends />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
