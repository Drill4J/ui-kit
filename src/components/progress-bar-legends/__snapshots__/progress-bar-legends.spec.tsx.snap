// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarLegends should match snapshot 1`] = `
<div>
  <div
    className="progress-bar-legends__Panel-sc-9uzeyp-0 kuTFXc"
  >
    <div
      className="progress-bar-legends__Legend-sc-9uzeyp-1 bkMnrW"
    >
      <div
        className="progress-bar-legends__Percentage-sc-9uzeyp-2 gKjXXj"
        type="start"
      >
        0
      </div>
    </div>
    <div
      className="progress-bar-legends__Legend-sc-9uzeyp-1 bkMnrW"
    >
      <div
        className="progress-bar-legends__Percentage-sc-9uzeyp-2 jwPEgy"
      >
        25
      </div>
    </div>
    <div
      className="progress-bar-legends__Legend-sc-9uzeyp-1 bkMnrW"
    >
      <div
        className="progress-bar-legends__Percentage-sc-9uzeyp-2 jwPEgy"
      >
        50
      </div>
    </div>
    <div
      className="progress-bar-legends__Legend-sc-9uzeyp-1 bkMnrW"
    >
      <div
        className="progress-bar-legends__Percentage-sc-9uzeyp-2 jwPEgy"
      >
        75
      </div>
    </div>
    <div
      className="progress-bar-legends__Legend-sc-9uzeyp-1 bkMnrW"
    >
      <div
        className="progress-bar-legends__Percentage-sc-9uzeyp-2 lipODO"
        type="end"
      >
        100
      </div>
    </div>
  </div>
</div>
`;
